code/tests/_game_test_puppeteer.dm

```diff
@@ -94,14 +94,24 @@
 	puppet.rejuvenate()
 
 /datum/test_puppeteer/proc/get_last_chatlog()
-	if(!(puppet.mind.key in GLOB.game_test_chats))
-		return FALSE
-	var/list/puppet_chat_list = GLOB.game_test_chats[puppet.mind.key]
+	var/list/puppet_chat_list = get_chatlogs()
 	return puppet_chat_list[length(puppet_chat_list)]
 
 /datum/test_puppeteer/proc/last_chatlog_has_text(snippet)
 	return findtextEx(get_last_chatlog(), snippet)
 
+/datum/test_puppeteer/proc/any_chatlog_has_text(snippet)
+	for(var/chat_line in get_chatlogs())
+		if(findtextEx(chat_line, snippet))
+			return TRUE
+
+	return FALSE
+
+/datum/test_puppeteer/proc/get_chatlogs()
```

Comment:
I feel like the `return FALSE` here isn't gonna do what you intend it to; I think you want to either return null or an empty list and return null from `get_last_chatlog()` instead

Reply:
Good catch, fixed